File: Lab5C#/Figure/Square.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Figures
{
    public class Square : Figure
    {
        public Square(double[] coordinates) : base(coordinates) {}

        public override double Perimeter()
        {
            return (a.VectorLength(b) + b.VectorLength(c)) * 2;
        }

        public override double Area()
        {
            return a.VectorLength(b) * b.VectorLength(c);
        }

        public void Output()
        {
            Console.Write("a = "); a.Output();
            Console.Write("b = "); b.Output();
            Console.Write("c = "); c.Output();
            Console.Write("d = "); d.Output();
            Console.WriteLine();
        }

        public void OutputPerimeter()
        {
            Console.Write("Perimeter = " + Perimeter() + "\n");
        }

        public void OutputArea()
        {
            Console.Write("Square = " + Area() + "\n");
        }
    }
}

File: Lab6C++/Figure/Square.cpp
#include "stdafx.h"
#include "Figure.h"

Square::Square(double ax, double ay, double bx, double by, double cx, double cy, double dx, double dy) : Figure(ax, ay, bx, by, cx, cy, dx, dy) {}

double Square::Perimeter()
{
	return (a.VectorLength(b) + b.VectorLength(c)) * 2;
}

double Square::Ssquare()
{
	return a.VectorLength(b) * b.VectorLength(c);
}

void Square::Output()
{
	std::cout << "a = "; a.Output();
	std::cout << "b = "; b.Output();
	std::cout << "c = "; c.Output();
	std::cout << "d = "; d.Output();
	std::cout << "\n";
}

void Square::OutputPerimeter()
{
	std::cout << "Perimeter = " << Square::Perimeter() << std::endl;
}

void Square::OutputSquare()
{
	std::cout << "Square = " << Square::Ssquare() << std::endl;
}